### PR TITLE
Rpc tweaks

### PR DESCRIPTION
**Pull request by jedirv**
This has two issues handled

- execution of command with args if specified in RpcConfig
- testing mode added for Viz where it won't send back state

### CODE REVIEW

File: ci/tests/viz/rpc-viz/src/entity/test.rs [hunk 1]
```diff
@@ -9,13 +12,136 @@ fn test_to_from_proto() {
...
 
     match IdEntity::from_proto(&proto) {
-        Err(err) => {
-            panic!(
-                "Could not decode proto just created: {}. Proto: {:?}",
-                err,
-                proto
-            )
-        }
+        Err(err) => panic!(
```
Comment (LinearZoetrope) [inline ci/tests/viz/rpc-viz/src/entity/test.rs]: Instead of this do

```rust
let proto = entity.to_proto().expect(
    &format!("Could not decode proto just created: {}. Proto {:?}", err, proto)
 );
```

File: ci/tests/viz/rpc-viz/src/entity/test.rs [hunk 2]
```diff
@@ -9,13 +12,136 @@ fn test_to_from_proto() {
...
+            "Could not decode proto just created: {}. Proto: {:?}",
+            err,
+            proto
+        ),
         Ok(got) => assert!(entity.fuzzy_eq(&got, FUZZY_EQ_THRESH)),
     }
 }
+// THIS TEST SHOULD NOT BE DELETED - COMMENTED OUT UNTIL I ADD BROWSER HANDLING
```
Comment (LinearZoetrope) [inline ci/tests/viz/rpc-viz/src/entity/test.rs]: I'd prefer to not merge a giant block of commented code, any chance you could put this test into a new branch and come back to it when you add the functionality to the browser?

File: common_protos/scaii.proto [hunk 3]
```diff
@@ -215,3 +216,15 @@ message SerializationResponse {
 	required bytes serialized = 1;
 	required SerializationFormat format = 2;
 }
+// Since Rpc waits for a response for each packet that is sent,
+// but in non-testing mode, Viz does not send back state,
+// we send back a UserCommand of the NONE variety as a simple Ack.  
+// EXPLAIN is a standin for when explanation info is requested.
+enum UserCommandType {
```
Comment (LinearZoetrope) [inline common_protos/scaii.proto]: You can do this inline

```proto
message UserCommand {
    enum UserCommandType {
        NONE = 0;
        EXPLAIN = 1;
    }
    required UserCommandType user_command_type = 1;
    // ...
}
```

File: common_protos/scaii.proto [hunk 4]
```diff
@@ -215,3 +216,15 @@ message SerializationResponse {
...
+// Since Rpc waits for a response for each packet that is sent,
+// but in non-testing mode, Viz does not send back state,
+// we send back a UserCommand of the NONE variety as a simple Ack.  
+// EXPLAIN is a standin for when explanation info is requested.
+enum UserCommandType {
+	NONE = 0;
+	EXPLAIN = 1;
+}
+message UserCommand {
+	required UserCommandType user_command_type = 1;
```
Comment (LinearZoetrope) [inline common_protos/scaii.proto]: maybe shorten this to just `command_type`? This name is a bit long and leaving off `user` doesn't make it any less clear

File: core/src/internal/rpc/mod.rs [hunk 5]
```diff
@@ -3,25 +3,74 @@ use scaii_defs::protos::{MultiMessage, ScaiiPacket};
 use scaii_defs::protos::{ModuleInit, RpcConfig};
 use std::error::Error;
 use std::net::{IpAddr, SocketAddr};
-
 use websocket::client::sync::Client;
 use websocket::stream::sync::TcpStream;
 
 use scaii_defs::protos::endpoint::Endpoint;
-
+use std::process::Child;
```
Comment (LinearZoetrope) [inline core/src/internal/rpc/mod.rs]: Instead, you can do `use std::process::{Child, Command};`

File: core/src/internal/rpc/mod.rs [hunk 6]
```diff
@@ -3,25 +3,74 @@ use scaii_defs::protos::{MultiMessage, ScaiiPacket};
...
+                rx.recv().unwrap();
+            });
+        }
+    }
 
+    let result = startup_module(&rpc_config);
+    tx.send(String::from("rpc_started")).unwrap();
+    result
+}
+// pub fn init_rpc_broken(rpc_config: RpcConfig) -> Result<LoadedAs, Box<Error>> {
```
Comment (LinearZoetrope) [inline core/src/internal/rpc/mod.rs]: Again, I'd prefer not to merge a big commented block of code. Can you move it to a new branch or delete it?

File: core/src/internal/rpc/mod.rs [hunk 7]
```diff
@@ -3,25 +3,74 @@ use scaii_defs::protos::{MultiMessage, ScaiiPacket};
...
+//     let result = startup_module(&rpc_config);
+//     tx.send(String::from("rpc_started")).unwrap();
+//     handle.join().unwrap();
+//     result
+// }
+fn startup_module(rpc_config: &RpcConfig) -> Result<LoadedAs, Box<Error>> {
     let client = connect(&rpc_config)?;
-
-    match rpc_config.init_as.init_as.ok_or_else::<Box<Error>, _>(|| {
+    let init_as = rpc_config.init_as.clone();
```
Comment (LinearZoetrope) [inline core/src/internal/rpc/mod.rs]: It's not a huge deal, but why are you cloning it?

File: core/src/internal/rpc/test.rs [hunk 8]
```diff
@@ -74,6 +71,57 @@ fn connect_attempt() {
     handle.join().unwrap();
 }
 
+
+// //this test launches a chrome tab so commented out so it won't run with usual tests
```
Comment (LinearZoetrope) [inline core/src/internal/rpc/test.rs]: Same "commented out code" comments as above.

File: viz/js/scaiiViz.js [hunk 9]
```diff
@@ -465,6 +476,18 @@ function handleViz(vizData) {
   renderState(masterEntities);
 
 }
+function buildEchoVizInitMultiMessage(vizInit) {
+  var returnScaiiPacket = new proto.scaii.common.ScaiiPacket;
+  returnScaiiPacket.setVizInit(vizInit);
+  var mm = buildReturnMultiMessageFromScaiiPacket(returnScaiiPacket);
+  return mm;
+}
+function buildMultiMessageWithUserCommand(userCommand) {
```
Comment (LinearZoetrope) [inline viz/js/scaiiViz.js]: Can you add a whitespace line between your function definitions? The lack of whitespace makes it a little hard to read this file. (Maybe leave this for a separate commit later, doesn't have to be right now).

**Comment by LinearZoetrope** [comment]
Also, could you move the testing mode to a new branch and make a separate PR?

File: ci/tests/viz/rpc-viz/src/entity/test.rs (hunk 1 above)
Comment (jedirv) [reply ci/tests/viz/rpc-viz/src/entity/test.rs]: This was one you wrote but I'll change :^)

File: ci/tests/viz/rpc-viz/src/entity/test.rs (hunk 2 above)
Comment (jedirv) [reply ci/tests/viz/rpc-viz/src/entity/test.rs]: Good idea - I'll do

File: common_protos/scaii.proto (hunk 3 above)
Comment (jedirv) [reply common_protos/scaii.proto]: will do

File: common_protos/scaii.proto (hunk 4 above)
Comment (jedirv) [reply common_protos/scaii.proto]: will do

File: core/src/internal/rpc/mod.rs (hunk 5 above)
Comment (jedirv) [reply core/src/internal/rpc/mod.rs]: ok

File: core/src/internal/rpc/mod.rs (hunk 6 above)
Comment (jedirv) [reply core/src/internal/rpc/mod.rs]: will do

File: ci/tests/viz/rpc-viz/src/entity/test.rs (hunk 1 above)
Comment (LinearZoetrope) [reply ci/tests/viz/rpc-viz/src/entity/test.rs]: I'm dumb

File: core/src/internal/rpc/mod.rs (hunk 7 above)
Comment (jedirv) [reply core/src/internal/rpc/mod.rs]: I think I was led there by an ownership complaint, but will check when repositioning the branches

File: core/src/internal/rpc/test.rs (hunk 8 above)
Comment (jedirv) [reply core/src/internal/rpc/test.rs]: will put in new branch

File: viz/js/scaiiViz.js (hunk 9 above)
Comment (jedirv) [reply viz/js/scaiiViz.js]: certainly

File: core/src/internal/rpc/mod.rs (hunk 7 above)
Comment (LinearZoetrope) [reply core/src/internal/rpc/mod.rs]: Unless this is mine, whatever

**Comment by jedirv** [comment]
I'll do all these changes as part of splitting this into two branches.

File: core/src/internal/rpc/mod.rs (hunk 7 above)
Comment (LinearZoetrope) [reply core/src/internal/rpc/mod.rs]: It's not a big deal. `init_as` isn't large so the clone is cheap, I just avoid `clone` where possible.
